### PR TITLE
[Discussion] [v6r19?] Initial attempt to replace pyGSI with M2Crypto.

### PR DESCRIPTION
**Pull request by miloszz**
At the moment it's mostly proof of concept that `pyGSI` can be replaced with `M2Crypto` in reasonable time.

What’s done:
- `dirac-proxy-init` and what is most essential for it to run
- changes in `X509*` classes (cleaning out some mess)
- awful hardcoded hacks in socket/protocol layer to avoid problems with modified X509* classes
- more awful hacks, marked more clearly now ;)

What works:
- `dirac-proxy-init`, also with `--rfc` that includes some pyGSI magic that hasn't been converted to M2Crypto

### CODE REVIEW

File: Core/Security/X509Chain.py [hunk 1]
```diff
@@ -147,34 +167,33 @@ def loadProxyFromString( self, pemData ):
...
-      blob = [ [ "1.3.6.1.5.5.7.21.1" ] ] if not rfcLimited else [ [ "1.3.6.1.4.1.3536.1.1.1.9" ] ]
-      asn1Obj = crypto.ASN1( blob )
-      asn1Obj[0][0].convert_to_object()
-      asn1dump = binascii.hexlify( asn1Obj.dump() )
-      extval = "critical,DER:" + ":".join( asn1dump[i:i + 2] for i in range( 0, len( asn1dump ), 2 ) )
-      ext = crypto.X509Extension( "proxyCertInfo", extval )
-      extList.append( ext )
-    return extList
+      if rfc:
+        ext =  M2Crypto.X509.new_extension( 'proxyCertInfo', 'critical, language:1.3.6.1.5.5.7.21.1', critical = 1 )
```
Comment (fstagni) [inline Core/Security/X509Chain.py]: I would love to see an explanation of this.

And in general, I would love to see some documentation. Would you provide, please? before we (and especially you) forget what's this all about.

**Comment by miloszz** [comment]
Aaaaah, and I forgot that rebase makes a mess.

**Comment by fstagni** [comment]
Ouf... I think you'd need to redo this PR! :-/

**Comment by chaen** [comment]
@miloszz is this the branch to be used, or do you have a clean PR somewhere ?

**Comment by miloszz** [comment]
I could make clean PR, but this one isn't messed up as it was before with rebasing (github changed something?). This is THE branch.

**Comment by chaen** [comment]
I am afraid it is messed up, it wants to remove some comments I added in v6r18 for example.

**Comment by miloszz** [comment]
Ok, I will take a look into this.

**Comment by miloszz** [comment]
I'm targeting integration with this PR, maybe that's the problem?

**Comment by atsareg** [comment]
This is OK to target the integration branch

**Comment by miloszz** [comment]
To be clear: X509Request is not done yet. I will focus on it now.

**Comment by miloszz** [comment]
I'm going to close it and open new one targeted at v6r18, as requested.

**Comment by chaen** [comment]
please don't
There was a misunderstanding.
Integration is correct

**Comment by miloszz** [comment]
Closed this PR *after* rebase. Recovered from backup, recreated branch. New PR on the way.

**Pull request closed without merging**